Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.

Patch rules_boost to:
  1. download boost 1.80.0 from the public boost archives (was: NVIDIA internal mirror)
  2. use @boringssl instead of @openssl for asio_ssl

diff --git boost/boost.bzl boost/boost.bzl
--- boost/boost.bzl
+++ boost/boost.bzl
@@ -214,13 +214,14 @@
         sha256 = "4b2136f98bdd1f5857f1c3dea9ac2018effe65286cf251534b6ae20cc45e1847",
         strip_prefix = "boost_1_80_0",
         urls = [
-            "https://boostorg.jfrog.io/artifactory/main/release/1.80.0/source/boost_1_80_0.tar.gz",
+            "https://archives.boost.io/release/1.80.0/source/boost_1_80_0.tar.gz",
+            "https://sourceforge.net/projects/boost/files/boost/1.80.0/boost_1_80_0.tar.gz",
         ],
     )
 
     maybe(
         http_archive,
-        name = "openssl",
+        name = "boringssl",
         sha256 = "6f640262999cd1fb33cf705922e453e835d2d20f3f06fe0d77f6426c19257308",
         strip_prefix = "boringssl-fc44652a42b396e1645d5e72aba053349992136a",
         url = "https://github.com/google/boringssl/archive/fc44652a42b396e1645d5e72aba053349992136a.tar.gz",
diff --git BUILD.boost BUILD.boost
--- BUILD.boost
+++ BUILD.boost
@@ -443,7 +443,7 @@
     visibility = ["//visibility:public"],
     deps = [
         ":asio",
-        "@openssl//:ssl",
+        "@boringssl//:ssl",
     ],
 )
 
